signature/policy_eval_signedby.go

```diff
@@ -70,6 +71,13 @@ func (pr *prSignedBy) isSignatureAuthorAccepted(image types.UnparsedImage, sig [
 			return PolicyRequirementError(fmt.Sprintf("Signature by key %s is not accepted", keyIdentity))
 		},
 		validateSignedDockerReference: func(ref string) error {
+			r := image.Reference().DockerReference()
+			if _, isCanonical := r.(reference.Canonical); isCanonical || reference.IsNameOnly(r) {
+				if !pr.UntaggedReference.matchesDockerReference(image, ref) {
```

Comment:
> That makes good sense, OTOH there will be various callers which just use an ImageSource and some of them could be badly surprised that a digest reference can give them a different image. (But then, something would still need to verify the digests of the manifest and layers, which ImageSource doesn’t do either. So, perhaps it really should be in image.Copy or something.)

you, I see image.Copy as a good fit for this check, people who use source(s) directly should know how the whole stuff works and if they're surprised they're getting another digest, well, it's their fault. Source(s) are low level primitives to interact with _registries_ and people using them should know that, we should make that clear in documentation.